### PR TITLE
Resync 'svg/text' tests from WPT upstream

### PR DESCRIPTION
**Pull request by Ahmad-S792**
#### 75f8bb0c2b8bbc26b237199283723ff219daff29
<pre>
Resync &apos;svg/text&apos; tests from WPT upstream

<a href="https://bugs.webkit.org/show_bug.cgi?id=262651">https://bugs.webkit.org/show_bug.cgi?id=262651</a>

Reviewed by Tim Nguyen.

This PR is to upstream all tests from &apos;svg/text&apos; directory based on WPT ToT hash:

Upstream hash: 5abf5b83de6107b57dcfab0c567ee857e05bc91e

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/gradient-after-reposition.html
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/gradient-after-reposition-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/gradient-after-reposition-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-dynamic.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-dynamic-ref.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-dynamic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-ref.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/multiple-textpaths.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/multiple-textpaths-ref.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/multiple-textpaths-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/no-background.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/no-background-ref.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/no-background-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/no-margin-border-padding.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/no-margin-border-padding-ref.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/no-margin-border-padding-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/opacity.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/opacity-ref.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/opacity-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-clipped-offscreen-move-onscreen-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-huge-font-size-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-transform-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-transform-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-transform-001.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-transform-002-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-transform-002-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/text-transform-002.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/transform-dynamic-change-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/transform-dynamic-change-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/transform-dynamic-change-root-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/transform-dynamic-change-root-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/transform-dynamic-change-root.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/transform-dynamic-change.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/writing-mode-dynamic-change-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/writing-mode-dynamic-change-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/writing-mode-dynamic-change.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute-dynamic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute-dynamic.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getextentofchar-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getextentofchar.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getrotationofchar-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getrotationofchar.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getstartpositionofchar-dominant-baseline-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getstartpositionofchar-dominant-baseline.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getsubstringlength-emoji-ligatures-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/getsubstringlength-emoji-ligatures.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/lengthadjust-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/lengthadjust.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/reattach-crash.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/textlength-inconsistent-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/textlength-inconsistent.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/transform-dynamic-update-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/transform-dynamic-update.html:
* LayoutTests/imported/w3c/web-platform-tests/svg/text/scripted/w3c-import.log:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-dynamic-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/multiple-textpaths-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/no-background-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/no-margin-border-padding-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/opacity-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute-dynamic-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/text/scripted/lengthadjust-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-dynamic-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/lang-attribute-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/multiple-textpaths-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/no-background-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/no-margin-border-padding-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/opacity-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute-dynamic-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/svg/text/reftests/xml-lang-attribute-expected.txt:

Canonical link: <a href="https://commits.webkit.org/268943@main">https://commits.webkit.org/268943@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/63cc6de1ef68952ae710dd4221ab5c919596b7f8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21057 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21437 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22124 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22939 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19593 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24691 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21623 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20845 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21281 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21016 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18266 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23793 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18174 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19094 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25371 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19255 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19295 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23300 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19840 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16860 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19110 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/18926 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5057 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23412 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19685 "Built successfully") | | |

### CODE REVIEW

**Comment by Ahmad-S792** [comment]
NOTE - this is my first try to import, so please be patient with me. :-)

File: LayoutTests/imported/w3c/resources/import-expectations.json [hunk 1]
```diff
@@ -414,6 +414,7 @@
     "web-platform-tests/subresource-integrity": "import",
     "web-platform-tests/svg": "import",
     "web-platform-tests/svg-aam": "skip",
+    "web-platform-tests/svg/text": "import",
```
Comment (nt1m) [inline LayoutTests/imported/w3c/resources/import-expectations.json]: This is implicit with `    "web-platform-tests/svg": "import",`. Can we undo this change?

Comment (Ahmad-S792) [reply LayoutTests/imported/w3c/resources/import-expectations.json]: Will do but will let EWS run so I can do all tweaks in one go. :-)

Comment (nt1m) [reply LayoutTests/imported/w3c/resources/import-expectations.json]: Sounds good!

Comment (aprotyas) [reply LayoutTests/imported/w3c/resources/import-expectations.json]: (isn't this autogenerated though?)

Comment (Ahmad-S792) [reply LayoutTests/imported/w3c/resources/import-expectations.json]: It is autogenerated. :-)

Comment (nt1m) [reply LayoutTests/imported/w3c/resources/import-expectations.json]: @aprotyas Yeah the script should probably be smarter about this (cc @gsnedders) but for now I'm catching these issues manually.

Comment (Ahmad-S792) [reply LayoutTests/imported/w3c/resources/import-expectations.json]: Done.

**Review by nt1m — APPROVED** [review]
The list of files in the commit message should omit `* LayoutTests/imported/w3c/resources/import-expectations.json:
`. But otherwise looks good, thanks!

**Comment by Ahmad-S792** [comment]
> The list of files in the commit message should omit `* LayoutTests/imported/w3c/resources/import-expectations.json: `. But otherwise looks good, thanks!

It is just in 'GitHub' issue, it is not in original commit message. 👍🏽
